src/faq/troubleshooting.rst

```diff
@@ -0,0 +1,222 @@
+If you're selling a model of hammer, and have 20 of them to sell, you might have
+documents with keys like hammer-1, hammer-2, etc, to represent each available
+hammer.
+
+Then, I'd create a view that gives me a list of available hammers, with a reduce
+function that lets me see a "total". These are completely off the cuff, but
+should give you an idea of what a working view would look like.
+
+**Map**::
+
+    function(doc)
+    {
+        if (doc.type == 'inventory_ticket' && doc.claimed_by == null ) {
+            emit(doc.product_key, { 'inventory_ticket' :doc.id, '_rev' : doc._rev });
+        }
+    }
+
+This gives me a list of available "tickets", by product key. I could grab a
+group of these when someone wants to buy a hammer, then iterate through sending
+updates (using the id and _rev) until I successfully claim one (previously
+claimed tickets will result in an update error).
+
+**Reduce**::
+
+    function (keys, values, combine) {
+        return values.length;
+    }
+
+This reduce function simply returns the total number of unclaimed
+inventory_ticket items, so you can tell how many "hammers" are available for
+purchase.
+
+**Caveats**
+
+This solution represents roughly 3.5 minutes of total thinking for the
+particular problem you've presented. There may be better ways of doing this!
+That said, it does substantially reduce conflicting updates, and cuts down on
+the need to respond to a conflict with a new update. Under this model, you won't
+have multiple users attempting to change data in primary product entry. At the
+very worst, you'll have multiple users attempting to claim a single ticket, and
+if you've grabbed several of those from your view, you simply move on to the
+next ticket and try again.
+
+.. note::
+    This FAQ entry was borrowed from
+    http://stackoverflow.com/questions/299723/can-i-do-transactions-and-locks-in-couchdb
+    with permission from the author
+
+Why does creating my view take so long?
+---------------------------------------
+
+There are a number of possible reasons:
+
+1. Your reduce function is not reducing the input data to a small enough output.
+See Introduction_to_CouchDB_views#reduce_functions for more details.
+
+2. If you have a lot of documents or lots of large documents (going into the
+millions and Gigabytes), the first time a view index is created just takes the
+time it is needed to run through all documents.
+
+3. If you use the emit()-function in your view with doc as the second parameter
+you effectively copy your entire data into the view index. This takes a lot of
+time. Consider rewriting your emit() call to emit(key, null); and query the view
+with the `?include_docs=true parameter` to get all doc's data with the view
+without having to copy the data into the view index.
+
+How can I "JOIN" with CouchDB?
+------------------------------
+
+See https://www.cmlenz.net/archives/2007/10/couchdb-joins
```

Comment:
We already have the entire View Cookbook for SQL Jockeys - we should link to that instead, assuming we even keep this section of the FAQ in here.